backend/src/routes/api/notebooks/utils.ts

```diff
@@ -27,12 +29,24 @@ export const getNotebookStatus = async (
   const notebookName = notebook?.metadata.name;
   let newNotebook: Notebook;
   if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
-    const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
-      fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
-      return undefined;
-    });
-    if (route) {
-      newNotebook = await patchNotebookRoute(fastify, route, namespace, notebookName).catch((e) => {
+    const enableServiceMesh = featureFlagEnabled(
+      getDashboardConfig().spec.dashboardConfig.disableServiceMesh,
+    );
+    let host: string;
+    if (enableServiceMesh) {
+      host = await getServiceMeshGwHost(fastify, namespace).catch((e) => {
+        fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
+        return undefined;
+      });
+    } else {
+      const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
+        fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
+        return undefined;
+      });
+      host = route?.spec.host;
+    }
+    if (host) {
+      newNotebook = await patchNotebookRoute(fastify, host, namespace, notebookName).catch((e) => {
```

Comment:
Just to add more context here, there are certain rules that can be applied to improve the code, some of them are just to improve the syntax others are to avoid side effects (such as mixing catches and awaits). Let me show you an example of this same function refactored to improve the sintax:

```javascript
export const getNotebookStatus = async (
  fastify: KubeFastifyInstance,
  namespace: string,
  name: string,
): Promise<{ notebook: Notebook; isRunning: boolean; podUID: string }> => {
  const notebook = await getNotebook(fastify, namespace, name);
  const hasStopAnnotation = !!notebook?.metadata.annotations?.['kubeflow-resource-stopped'];
  const [isRunning, podUID] = hasStopAnnotation
    ? [false, '']
    : await listNotebookStatus(fastify, namespace, name);

  const notebookName = notebook?.metadata.name;

  if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
    const enableServiceMesh = featureFlagEnabled(
      getDashboardConfig().spec.dashboardConfig.disableServiceMesh,
    );
    const getHost = enableServiceMesh ? getServiceMeshGwHost : getRoute;
    const logError = (e: Error) =>
      fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);

    try {
      const hostObj = await getHost(fastify, namespace, notebookName);
      if (typeof hostObj === 'object' && hostObj !== null && 'spec' in hostObj) {
        const host = hostObj.spec.host;
        if (host) {
          try {
            const newNotebook = await patchNotebookRoute(fastify, host, namespace, notebookName);
            return { notebook: newNotebook, isRunning, podUID };
          } catch (e) {
            fastify.log.warn(`Failed patching route to notebook ${notebookName}: ${e.message}`);
            return { notebook, isRunning, podUID };
          }
        }
      }
    } catch (e) {
      logError(e);
    }
  }
};
```

Take the logic with a grain of salt, it was just a quick refactor to display some techniques (avoiding using a lot of mutable variables, leveraging some logic to typescript's types and more)